pkg/net/net.go

```diff
@@ -151,3 +151,15 @@ type BroadcastChannel interface {
 // public key as its argument and returns true if the message should be
 // processed or false otherwise.
 type BroadcastChannelFilter func(*ecdsa.PublicKey) bool
+
+// Firewall represents a set of rules the remote peer has to conform to so that
+// a connection with that peer can be approved.
+type Firewall interface {
```

Comment:
If we defined `Firewall` interface in `pkg/net` shouldn't the implementation be placed under `pkg/net/firewall` instead of `pkg/firewall`

Reply:
It was my initial plan but then I realized we don't want `net` to depend on `chain`. Right now, `net` defines `Firewall` interface so it doesn't care what's behind it.